src/main/java/org/apache/cassandra/sidecar/exceptions/ThrowableUtils.java

```diff
@@ -101,12 +102,16 @@ public static Throwable getCause(Throwable throwable, Predicate<Throwable> predi
                 return cause;
             }
 
-            fastTracer = getCause(fastTracer, 2);
-            if (cause == fastTracer && stop == null)
+            if (stop == null)
```

Comment:
another optimization 
```suggestion
            if (fastTracer != null && predicate.test(fastTracer))
            {
                return fastTracer;
            }

            if (stop == null)
```

Reply:
It is probably not. Because it adds another test of the predicate.